src/example_documentation.cpp

```diff
@@ -0,0 +1,128 @@
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief  This class serves as an example of how classes in libgdf should 
+ * be documented.
+ */
+/* ----------------------------------------------------------------------------*/
+template <typename T>
+class example_class
+{
+
+  // DESCRIPTION: Trivial class functions should be given names that make their purpose clear
+  // DESCRIPTION: If their name makes the functionality obvious, no documentation is necessary
+  void get_my_int() {
+    return my_int;
+  }
+  void set_my_int(int new_value){
+    my_int = new_value;
+  }
+
+  // DESCRIPTION: Non-trivial member functions should have a brief description of the
+  // function as well as all of its parameters. Every parameter should be decorated to 
+  // indicate if it is an input or output parameter, or both with @Param[in], @Param[out],
+  // and @Param[in,out] respectively.
+  /* --------------------------------------------------------------------------*/
+  /** 
+   * @brief This is a complicated function that requires more detailed documentation.
+   * 
+   * @Param[in] first_parameter  This parameter is an input parameter to the function
+   * @Param[in,out] second_parameter This parameter is used both as an input and output
+   * @Param[out] third_parameter This parameter is an output of the function
+   * 
+   * @Returns The result of the complex function
+   */
+  /* ----------------------------------------------------------------------------*/
+  T complicated_function(int const first_parameter,
+                         double * second_parameter,
+                         float * third_parameter)
+  {
+    // DESCRIPTION: Notice the use of *human readable* variable names. Human readable 
+    // variable names are vastly prefered to short, hard to read names. E.g.,
+    // use 'first_parameter' instead of 'fp'. When in doubt, opt for the longer, easier
+    // to read name that conveys the meaning and purpose of the variable. As developers,
+    // we usually spend more time reading code than writing code, so the easier you make it to
+    // read your code, the more efficient we will all be.
+    
+   
+    // DESCRIPTION: In-line comments that describe the logic inside of your functions
+    // are extremely helpful both to others as well as your future self to aid in 
+    // understanding your thought process
+  }
+
+
+private:
+  int my_int;                //< An example private member variable
+  std::vector<T> my_vector;  //< An example private member variable
+};
+
+// DESCRIPTION: Free functions should be commented in the same way as non-trivial
+// class member functions. If the function is templated, use @tparam to describe
+// the purpose of the template parameters.
+/* --------------------------------------------------------------------------*/
+/** 
+ * @Synopsis  An example of a free function (non-class member). This function
```

Comment:
Good catch. The reason is I updated my Vim plugin to do @brief instead of @Synopsis to be in-line with the document you created, but forget to update the entries I had already created :)

Reply:
Maybe that document should be a public part of libgdf?